python/graphstorm/model/edge_decoder.py

```diff
@@ -1071,6 +1095,403 @@ def get_relembs(self):
         """
         return self._w_relation.weight, self.etype2rid
 
+class LinkPredictRotatEDecoder(LinkPredictMultiRelationLearnableDecoder):
+    r""" Decoder for link prediction using the RotatE as the score function.
+
+        Score function of RotateE measures the angular distance between
+        head and tail elements and is defined as:
+
+        .. math:
+
+            d_r(h, t)=\|h\circ r-t\|
+
+        Its score function is defined as:
+
+        .. math:
+
+            gamma - \|h\circ r-t\|^2
+
+        where gamma is a margin.
+
+        More detials please refer to https://arxiv.org/abs/1902.10197
+        or https://dglke.dgl.ai/doc/kg.html#rotatee.
+
+    Parameters
+    ----------
+    etypes: list of tuples
+        The canonical edge types of the graph in the format of
+        [(src_ntype1, etype1, dst_ntype1), ...]
+    h_dim: int
+        The input dimension size. It is the dimension for both source and destination
+        node embeddings.
+    gamma: float
+        The gamma value for model initialization and score function. Default: 40.
```

Comment:
This class use `4.0` for the `gamma`. Should we keep all `gamma` be `40.`?

Reply:
No. Different decoder (score func) has different gamma value. 40. for RotatE is too large by default.